src/App.tsx

```diff
@@ -111,6 +115,20 @@ const App = () => {
     }
   }, [addresses.length, addressesStatus, dispatch, network.status])
 
+  useEffect(() => {
+    const interval = setInterval(() => {
+      if (pendingTxHashes.length > 0) {
+        dispatch(syncAddressesData(pendingTxHashes))
+      } else {
+        clearInterval(interval)
+      }
+    }, 2000)
+
+    return () => {
+      clearInterval(interval)
+    }
+  }, [pendingTxHashes, dispatch])
+
```

Comment:
FYI, I've already updated this in the next branch to use `useInterval`

Reply:
Perfect.